### ISSUE TITLE
Detaching a Grid with a frozen Column causes a client side Exception (Vaadin 8.0.0)

### ISSUE THREAD

**Issue by ssindelar** [vaadin/framework#8712]
If you have a Grid with a frozen column and the grid becomes detached (e.g. closing a window or changing a view) you get client side exception.
I compiled the Widgetset with "DETAILED" to get the most info:
```
count must be between 0 and the current number of columns (0)
java.lang.IllegalArgumentException: count must be between 0 and the current number of columns (0)
   at Unknown.java_lang_Throwable_Throwable__Ljava_lang_String_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.java_lang_Exception_Exception__Ljava_lang_String_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.java_lang_RuntimeException_RuntimeException__Ljava_lang_String_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.java_lang_IllegalArgumentException_IllegalArgumentException__Ljava_lang_String_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_widgets_Escalator$ColumnConfigurationImpl_setFrozenColumnCount__IV(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_widgets_Grid_updateFrozenColumns__V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_widgets_Grid_removeColumnSkipSelectionColumnCheck__Lcom_vaadin_client_widgets_Grid$Column_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_widgets_Grid_removeColumn__Lcom_vaadin_client_widgets_Grid$Column_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_connectors_grid_GridConnector_removeColumn__Lcom_vaadin_client_connectors_grid_ColumnConnector$CustomColumn_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_connectors_grid_ColumnConnector_onUnregister__V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_ConnectorMap_unregisterConnector__Lcom_vaadin_client_ServerConnector_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_communication_MessageHandler$3_unregisterRemovedConnectors__Lcom_vaadin_client_FastStringSet_2I(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_communication_MessageHandler$3_execute__V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_ApplicationConfiguration_runWhenDependenciesLoaded__Lcom_google_gwt_user_client_Command_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_communication_MessageHandler_handleJSON__Lcom_vaadin_client_ValueMap_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_communication_MessageHandler_handleMessage__Lcom_vaadin_client_ValueMap_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_vaadin_client_communication_XhrConnection$XhrResponseHandler_onResponseReceived__Lcom_google_gwt_http_client_Request_2Lcom_google_gwt_http_client_Response_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_google_gwt_http_client_Request_fireOnResponseReceived__Lcom_google_gwt_http_client_RequestCallback_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_google_gwt_http_client_RequestBuilder$1_onReadyStateChange__Lcom_google_gwt_xhr_client_XMLHttpRequest_2V(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.onreadystatechange<(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_google_gwt_core_client_impl_Impl_apply__Ljava_lang_Object_2Ljava_lang_Object_2Ljava_lang_Object_2Ljava_lang_Object_2(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_google_gwt_core_client_impl_Impl_entry0__Ljava_lang_Object_2Ljava_lang_Object_2Ljava_lang_Object_2Ljava_lang_Object_2(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.com_google_gwt_core_client_impl_Impl_entry__Lcom_google_gwt_core_client_JavaScriptObject_2Lcom_google_gwt_core_client_JavaScriptObject_2/<(net.ibhks.EuroSdsWidgetset-0.js)
   at Unknown.anonymous(Unknown)
```

Example for reproduction:

```
public class TestUi extends UI {
	private static final long serialVersionUID = 6146785985917164310L;

	@Override
	protected void init(VaadinRequest request) {
		addWindow(new GridWindow());
	}

	private static class GridWindow extends Window {

		private static final long serialVersionUID = -5109873553420121929L;
		private final Grid<String> grid = new Grid<>();

		public GridWindow() {
			setCaption("Test Window");
			setContent(grid);
			grid.setCaption("Grid");
			grid.addColumn(r -> "novalue");
			grid.setItems("item1", "item2");
			grid.setFrozenColumnCount(1);
		}
	}
}
```
Just close the window to get the error.

**Comment by hero1979** [comment]
Can someone from Vaadin team mark this as a bug ?
Discussed here https://vaadin.com/forum#!/thread/15323554

**Comment by Artur-** [comment]
Does this work in 8.0.3?

**Comment by ssindelar** [comment]
Seems to work with 8.1

**Comment by hesara** [comment]
Marking as fixed in latest 8.1 - probably fixed already earlier.
